src/script/components/ModalComponent.tsx

```diff
@@ -133,28 +130,32 @@ const ModalComponent: React.FC<ModalComponentProps> = ({
     timeoutId = window.setTimeout(() => {
       setDisplayNone(true);
       onClosed();
-    }, 150);
+    }, CLOSE_DELAY);
 
     return () => {
       window.clearTimeout(timeoutId);
     };
   }, [isShown]);
 
+  if (displayNone) {
+    return null;
+  }
+
   return (
+    // eslint-disable-next-line jsx-a11y/click-events-have-key-events,jsx-a11y/no-noninteractive-element-interactions
     <div
+      role="dialog"
+      aria-modal="true"
       onClick={onBgClick}
+      id={id}
       css={hasVisibleClass ? ModalOverlayVisibleStyles : ModalOverlayStyles}
       style={{display: displayNone ? 'none' : 'flex'}}
-      tabIndex={0}
-      role="button"
-      onKeyDown={noop}
-      id={id}
       className={className}
       {...rest}
     >
-      {showLoading ? (
-        <Icon.Loading width="48" height="48" css={{path: {fill: 'var(--modal-bg)'}}} />
-      ) : (
+      {showLoading && <Icon.Loading width="48" height="48" css={{path: {fill: 'var(--modal-bg)'}}} />}
+
+      {!showLoading && !displayNone && (
```

Comment:
```suggestion
      {!showLoading &&  (
```
Since you return early when it's `true`